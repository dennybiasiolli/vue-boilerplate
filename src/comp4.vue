<template>
  <div>
    <ol>
      <li v-for="(todo, key) in todos" :key="key">{{ todo.text }}</li>
    </ol>
  </div>
</template>

<script>
export default {
  data() {
    return {
      todos: [
        { text: 'Learn JavaScript' },
        { text: 'Learn Vue' },
        { text: 'Build something awesome' }
      ]
    }
  }
}
</script>
