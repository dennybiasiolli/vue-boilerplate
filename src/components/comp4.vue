<template>
  <v-list>
    <v-list-tile
      v-for="(todo, key) in todos"
      :key="key"
      @click="()=>{}"
    >
      <v-list-tile-action>
        {{ key + 1 }}
      </v-list-tile-action>
      <v-list-tile-content>
        <v-list-tile-title>{{ todo.text }}</v-list-tile-title>
        <v-list-tile-sub-title>{{ todo.text }}</v-list-tile-sub-title>
      </v-list-tile-content>
    </v-list-tile>
  </v-list>
</template>

<script>
export default {
  data() {
    return {
      todos: [
        { text: 'Learn JavaScript' },
        { text: 'Learn Vue' },
        { text: 'Build something awesome' }
      ]
    }
  }
}
</script>
