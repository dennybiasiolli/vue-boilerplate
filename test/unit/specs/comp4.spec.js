import { shallow } from 'vue-test-utils'
import comp4 from '@/comp4'

describe('comp4', () => {
  test('should match snapshot', (done) => {
    const component = shallow(comp4)
    expect(component.element).toMatchSnapshot()
    component.vm.todos.push({ text: 'test 4th element' })
    component.vm.$nextTick(() => {
      expect(component.element).toMatchSnapshot()
      done()
    })
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct default data', () => {
    expect(typeof comp4.data).toBe('function')
    const defaultData = comp4.data()
    expect(defaultData.todos).toEqual([
      { text: 'Learn JavaScript' },
      { text: 'Learn Vue' },
      { text: 'Build something awesome' }
    ])
  })

  // Inspect the component instance on mount
  test('correctly sets the todos array when created', () => {
    const vm = shallow(comp4).vm
    expect(vm.todos).toEqual([
      { text: 'Learn JavaScript' },
      { text: 'Learn Vue' },
      { text: 'Build something awesome' }
    ])
  })

  // Mount an instance and inspect the render output
  test('renders the correct todos list', () => {
    const vm = shallow(comp4).vm
    expect(vm.$el.textContent).toContain('Build something awesome')
    expect(vm.$el.querySelector('ol').childElementCount).toBe(3)
  })
})
