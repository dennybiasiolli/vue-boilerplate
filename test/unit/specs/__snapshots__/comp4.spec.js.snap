// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comp4 should match snapshot 1`] = `
<div>
  <ol>
    <li>
      Learn JavaScript
    </li>
    <li>
      Learn Vue
    </li>
    <li>
      Build something awesome
    </li>
  </ol>
</div>
`;

exports[`comp4 should match snapshot 2`] = `
<div>
  <ol>
    <li>
      Learn JavaScript
    </li>
    <li>
      Learn Vue
    </li>
    <li>
      Build something awesome
    </li>
    <li>
      test 4th element
    </li>
  </ol>
</div>
`;
